ingest/src/server_mobile.rs

```diff
@@ -150,10 +169,25 @@ impl GrpcServer {
             .map_err(|_| Status::invalid_argument("invalid signature"))?;
         Ok((public_key, event))
     }
+
+    async fn verify_known_carrier_key(&self, public_key: PublicKey) -> VerifyResult<()> {
```

Comment:
This is a pretty substantial change for our ingestors as it's effectively performing a verification on the incoming data when we have traditionally done this kind of check entirely in the verifiers with the one exception of signature verification on the incoming request message.

After our experience with the GRPC request/response needed to submit disco mapping verified reports to the ETL this feels like we're setting ourselves up for a similar pain in that _every_ ingested report that needs this validation will require a GRPC request and response to the mobile config service at a minimum and if the carrier key (thankfully a small set at the moment) falls out of the cache it means a db query by the mobile config service in order service the request as well.

What's the drive here behind performing this check in the ingestor?

Reply:
The reasoning behind this was to avoid having to push a message to the verifier just to have the verifier carry out this verification when the verifier itself does nothing with the said message...other than consume it, check the key and write it back out with another wrapper.  

The check does not have to be used on every message, I would consider it intent for only those message types which are generated by a non public service but where the message itself is required to hit a public S3 bucket ( which is the case here ).